theano/scalar/basic.py

```diff
@@ -3843,7 +3843,7 @@ def __init__(self, inputs, outputs):
         self.nin = len(inputs)
         self.nout = len(outputs)
         self.init_fgraph()       # self.fgraph
-
+        self.init_c_code()
```

Comment:
I get errors after removing that change(this branch has been rebased with yesterday's master)

```
theano.gof.opt: ERROR: Optimization failure due to: local_pow_specialize_device
theano.gof.opt: ERROR: node: Elemwise{pow,no_inplace}(x, TensorConstant{(1,) of 6})
theano.gof.opt: ERROR: TRACEBACK:
theano.gof.opt: ERROR: Traceback (most recent call last):
  File "/Users/Ramana/projects/theano/Theano/theano/gof/opt.py", line 1962, in process_node
    remove=remove)
  File "/Users/Ramana/projects/theano/Theano/theano/gof/toolbox.py", line 391, in replace_all_validate_remove
    chk = fgraph.replace_all_validate(replacements, reason)
  File "/Users/Ramana/projects/theano/Theano/theano/gof/toolbox.py", line 340, in replace_all_validate
    fgraph.replace(r, new_r, reason=reason, verbose=False)
  File "/Users/Ramana/projects/theano/Theano/theano/gof/fg.py", line 508, in replace
    self.change_input(node, i, new_r, reason=reason)
  File "/Users/Ramana/projects/theano/Theano/theano/gof/fg.py", line 450, in change_input
    self.__import_r__(new_r, reason=reason)
  File "/Users/Ramana/projects/theano/Theano/theano/gof/fg.py", line 351, in __import_r__
    self.__import__(variable.owner, reason=reason)
  File "/Users/Ramana/projects/theano/Theano/theano/gof/fg.py", line 414, in __import__
    self.execute_callbacks('on_import', node, reason)
  File "/Users/Ramana/projects/theano/Theano/theano/gof/fg.py", line 588, in execute_callbacks
    fn(self, *args, **kwargs)
  File "/Users/Ramana/projects/theano/Theano/theano/gof/opt.py", line 561, in on_import
    self.process_node(fgraph, node)
  File "/Users/Ramana/projects/theano/Theano/theano/gof/opt.py", line 675, in process_node
    if inputs_match and node.op == candidate.op:
  File "/Users/Ramana/projects/theano/Theano/theano/gof/utils.py", line 196, in __eq__
    tuple(getattr(other, a) for a in props))
  File "/Users/Ramana/projects/theano/Theano/theano/scalar/basic.py", line 3967, in __eq__
    return (self._c_code == other._c_code)
AttributeError: 'Composite' object has no attribute '_c_code'
```